### ISSUE TITLE
Clarify pull request branch language

### ISSUE THREAD

**Issue by davidmcguiredesign** [github/docs#35561]
### Code of Conduct

- [X] I have read and agree to the GitHub Docs project's [Code of Conduct](https://github.com/github/docs/blob/main/.github/CODE_OF_CONDUCT.md)

### What article on docs.github.com is affected?

- [Glossary](https://docs.github.com/en/get-started/learning-about-github/github-glossary)
- [Resolving a merge conflict on GitHub](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/addressing-merge-conflicts/resolving-a-merge-conflict-on-github)
- [About pull request merges](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/incorporating-changes-from-a-pull-request/about-pull-request-merges)

### What part(s) of the article would you like to see updated?

TLDR: **To curtail confusion, identify the branch that a pull request will _change_.**

### The problem

I and many others find pull requests confusing to learn, partly because the terms used to discuss it often conflict (ironically) with their intuitive meanings. Here are ways the GitHub Docs and web UI describe the two branches in a pull request, and intuitive issues with each:

1. **`HEAD`** conflicts with the stream metaphor used by "upstream" and "downstream" branches. The head of a stream is the point _from_ which it flows, but in git-land, it means "you are here," which is usually (but not always!) the tip—the point _to_ which the project has flowed—the _ahead_ point. 

2. **`base`** could intuitively mean "the branch onto which the other one is tacked," or "the branch from which the other came" (actually just meaning the _older_ branch), or "the reference branch that doesn’t change".

3. "**target** branch" could intuitively mean the "branch that has the desired state," or the "branch being changed". (Conversely, "**source** branch" could intuitively mean the _older_ branch—like the stream metaphor—which is usually the one being changed.)

4. The meaning of "incoming" depends on where I’m standing. It’s fine as long as I remember that when approving a pull request, I stand at the branch being changed, but it still can be counter-intuitive after spending all day working on the other branch. 

5. "merge **into**", "rebase **onto**", and the arrow icons are intuitive as long as I accept the metaphor that the sentence’s direct object, the "stationary" branch, is the one that will be changed—the moving part. (Is this a nuance? Yes, but so is a checkbox for a negative, like "✅ Do not enforce strict types". Is this an _important_ nuance? I’m open to debate!) 

And finally, the point that brought me here: 

6. The **pull** page uses \#5 as described above (the branch with the desired state moving _to_ the branch that will change), but the **conflicts** page (and its merge page) reverses the order. The unenlightened (myself included) could easily read the apparently reversed order as "direct the changing branch to 'look _to_' the correct branch," forgetting that the resolution needs its own commit, distinct from the merge commit.

Pull requests are a GitHub feature, so GitHub should be the authoritative voice. The inconsistency weakens its authority. 

### But is there a solution here?

Considering all the nuances, I think the solution is pretty simple. **The first time a document specifies one of the two branches, it should note whether it is or is not the branch that the pull request will change.** 

A little more context on the **conflicts** page would be welcome too: "Commit conflict resolution to `dev` to allow pulling `dev` into `main`". I recognize that this is outside the scope of the Docs repository, but I hope a Hubber will notice and like the suggestion enough to escalate it.

**Comment by nguyenalex836** [comment]
@davidmcguiredesign Thank you for raising this issue! I'll get this triaged for review :sparkles: Our team will provide feedback regarding the best next steps for this issue - thanks for your patience! 💛

**Comment by nguyenalex836** [comment]
@davidmcguiredesign Thank you for your patience while our team reviewed! 

We are constantly reviewing our docs and have plans to improve the content around pull requests, and we’d prefer not to pre-empt that at this time 💛 We appreciate your willingness to improve our docs!
